### PR TITLE
fix(Core/Script): Molten Core: Lucifron

### LINKED ISSUE #5456: [Molten Core] Lucifron
### [Molten Core] Lucifron

This issue is meant to track the issues with the Molten Core Boss "Lucifron"

## CURRENT BEHAVIOUR:

grafik

## EXPECTED BLIZZLIKE BEHAVIOUR:

- Correct timers

## SOURCE:

- Research done by @T1ti 

## STEPS TO REPRODUCE THE PROBLEM:

1. Fight Lucifron

## AC HASH/COMMIT:

d11d021f38e0e0db27e7c74da801402e7e5f17d9

## OPERATING SYSTEM:

- Universal

## MODULES:

- None

## CUSTOMIZATIONS:

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/98101614-molten-core-lucifron?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by Hacki95**
Co-Author: @sanctum32 (Code)
Co-Author: @T1ti (Research)
Co-Author: @Hacki95 (Coordination)

## Changes Proposed:
- Correct all timers to the researched values.
- Randomize the timers.

## Issues Addressed:

- Closes #5456

## SOURCE:

- Research in #5456

## Tests Performed:

- Fix applied
- Fought agains Lucifron

## How to Test the Changes:

1. Apply the Fix
2. .tele moltencore
3. Fight Lucifron
4. Check the timers and spells.

## Known Issues and TODO List:

- /

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_lucifron.cpp [hunk 1]
```diff
@@ -41,42 +40,50 @@ class boss_lucifron : public CreatureScript
         {
         }
 
-        void EnterCombat(Unit* victim) override
+        void EnterCombat(Unit* /*attacker*/) override
```
Comment (Kitzunu) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_lucifron.cpp:43]: Why?

File: src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_lucifron.cpp [hunk 2]
```diff
@@ -93,4 +100,4 @@ class boss_lucifron : public CreatureScript
 void AddSC_boss_lucifron()
 {
     new boss_lucifron();
-}
+}
```
Comment (Kitzunu) [inline src/server/scripts/EasternKingdoms/BlackrockMountain/MoltenCore/boss_lucifron.cpp:103]: Newline

**Comment by claudiodfc** [comment]
## Tested this on Windows 10.
### Commands
.go c id 12118

### Test output
✔️ Lucifron casts Shadow Shock spell every 5 seconds.
✔️ Lucifron casts Impending Doom spell every 6 to 11 seconds.
✔️ Lucifron casts Lucifron Curse spell every 11 to 14 seconds.

**Comment by claudiodfc** [comment]
@FrancescoBorzi Ready to be merged

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
good job @Hacki95 and @claudiodfc

**Pull request merged**